cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -73,6 +76,18 @@ public static String getNamespacedBaseDir(NamespacedLocationFactory namespacedLo
     return namespacedLocationFactory.get(namespaceId.toId()).append(logBaseDir).toString();
   }
 
+  public static Location getNamespacedBaseDirLocation(final NamespacedLocationFactory namespacedLocationFactory,
+                                                      final String logBaseDir, final NamespaceId namespaceId,
+                                                      Impersonator impersonator) throws Exception {
+    Preconditions.checkArgument(logBaseDir != null, "Log Base dir cannot be null");
+    return impersonator.doAs(namespaceId, new Callable<Location>() {
+      @Override
+      public Location call() throws Exception {
+        return namespacedLocationFactory.get(namespaceId.toId()).append(logBaseDir);
```

Comment:
It doesn't do any file operation, but it creates a Location object, which is when impersonation is needed (because this is when the underlying hadoop FileContext object is created).